pkg/cri/resource-manager/events.go

```diff
@@ -109,21 +110,22 @@ func (m *resmgr) SendEvent(event interface{}) error {
 func (m *resmgr) processEvent(e interface{}) {
 	evtlog.Debug("received event of type %T...", e)
 
-	m.Lock()
-	defer m.Unlock()
-
 	switch event := e.(type) {
 	case string:
 		evtlog.Debug("'%s'...", event)
-	case *metrics.Event:
+	case *events.Metrics:
+		m.Lock()
```

Comment:
We should maybe have a convention about locking: should it be done here in the generic event handler or in the functions themselves (as it is done in `DeliverPolicyEvent()`).

Reply:
Yes. It probably will need to be 'functions lock as needed'.

Reply:
Updated processAvx() to lock for consistency.